src/c++/vdo/base/slab-depot.c

```diff
@@ -2262,23 +2269,25 @@ static void load_reference_block_endio(struct bio *bio)
 }
 
 /**
- * load_reference_block() - After a block waiter has gotten a VIO from the VIO pool, load the
- *                          block.
- * @waiter: The waiter of the block to load.
+ * load_reference_block_set() - After a block waiter has gotten a VIO from the VIO pool, load
+ *                              the run of blocks.
+ * @waiter: The waiter of the first block to load.
  * @context: The VIO returned by the pool.
  */
-static void load_reference_block(struct vdo_waiter *waiter, void *context)
+static void load_reference_block_run(struct vdo_waiter *waiter, void *context)
 {
 	struct pooled_vio *pooled = context;
 	struct vio *vio = &pooled->vio;
 	struct reference_block *block =
 		container_of(waiter, struct reference_block, waiter);
 	size_t block_offset = (block - block->slab->reference_blocks);
+	int max_block_count = block->slab->reference_block_count - block_offset;
+	int num_blocks = min_t(int, vio->block_count, max_block_count);
```

Comment:
if block_offset is a number of blocks (which it appears to be) can we change its type to block_count_t, or possibly u32 or u16? size_t is supposed to count bytes. (slab->reference_block_count is already a u32, maybe use that.)

If you don't inline it, "remainder" or "remaining_blocks" feels more descriptive than "max_block_count" (max of what?) The variable could also be inlined, which would eliminate the naming issue. 

In my opinion "block_count" is always a superior name to "num_blocks". (Because again, "num " is not a word. The only reason I didn't change num_foo everywhere is that Tom really like it so it's all over UDS.)